src/Shared/IConfigurationExtensions.cs

```diff
@@ -8,6 +8,46 @@ namespace Aspire;
 
 internal static class IConfigurationExtensions
 {
+    public static T GetValue<T>(this IConfiguration configuration, string primaryKey, string secondaryKey, T defaultValue)
+    {
+        var primaryValue = configuration.GetValue(typeof(T), primaryKey, null);
+        if (primaryValue is not null)
+        {
+            return (T)primaryValue;
+        }
+
+        var secondaryValue = configuration.GetValue(typeof(T), secondaryKey, null);
+        if (secondaryValue is not null)
+        {
+            return (T)secondaryValue;
+        }
+
+        return defaultValue;
+    }
+
+    public static bool? GetBool(this IConfiguration configuration, string primaryKey, string secondaryKey)
+    {
+        var value = configuration.GetBool(primaryKey) ?? configuration.GetBool(secondaryKey);
+        return value;
+    }
+
+    public static string? GetString(this IConfiguration configuration, string primaryKey, string secondaryKey, bool fallbackOnEmpty = false)
+    {
+        var primaryValue = configuration.GetValue(typeof(string), primaryKey, null);
+        if (primaryValue is not null && !fallbackOnEmpty || primaryValue is string { Length: > 1 })
```

Comment:
This was skipping strings with 1 character.
```suggestion
        if (primaryValue is not null && !fallbackOnEmpty || primaryValue is string { Length: > 0 })
```

Reply:
doh, thanks